src/access_nri_intake/data/utils.py

```diff
@@ -48,21 +48,41 @@ def available_versions(pretty: bool = True):
     # Work out where the catalogs are stored
     base_path = _get_catalog_rp()
 
+    # Grab the extant catalog and work out its min and max versions
+    try:
+        with open(get_catalog_fp()) as cat:
+            cat_yaml = yaml.safe_load(cat)
+            vers_min = cat_yaml["sources"]["access_nri"]["parameters"]["version"]["min"]
+            vers_max = cat_yaml["sources"]["access_nri"]["parameters"]["version"]["max"]
+            vers_def = cat_yaml["sources"]["access_nri"]["parameters"]["version"][
+                "default"
+            ]
+    except FileNotFoundError:
+        raise FileNotFoundError(f"Unable to find catalog at {get_catalog_fp()}")
+    except KeyError:
+        raise RuntimeError(f"Catalog at {get_catalog_fp()} not correctly formatted")
+
     # Grab all the catalog names
-    cats = [d for d in os.listdir(base_path) if re.search(CATALOG_NAME_FORMAT, d)]
+    cats = [
+        d.name
+        for d in base_path.iterdir()
+        if re.search(CATALOG_NAME_FORMAT, d.name)
+        and d.is_dir()
+        and ((d.name >= vers_min and d.name <= vers_max) or d.name == vers_def)
+    ]
     cats.sort(reverse=True)
 
     # Find all the symlinked versions
-    symlinks = [s for s in cats if os.path.islink(os.path.join(base_path, s))]
+    symlinks = [s for s in cats if (Path(base_path) / s).is_symlink()]
 
-    symlink_targets = {
-        s: os.path.basename(os.readlink(os.path.join(base_path, s))) for s in symlinks
-    }
+    symlink_targets = {s: (base_path / s).readlink().name for s in symlinks}
 
     if pretty:
-        for i, c in enumerate(cats):
+        for c in cats:
             if c in symlink_targets.keys():
                 c += f"(-->{symlink_targets[c]})"
+            if c == vers_def:
+                c += "*"
             print(c)
         return
 
```

Comment:
@charles-turner-1 having made the suggested changes, I'm now getting a `mypy` pre-commit error on this line:

```
src/access_nri_intake/data/utils.py:86: error: Unsupported left operand type for + ("TextIOWrapper[_WrappedBuffer]")  [operator]
src/access_nri_intake/data/utils.py:88: error: Unsupported left operand type for + ("TextIOWrapper[_WrappedBuffer]")  [operator]
```

Any hints? I'm not sure what's caused it to suddenly appear... the tests all pass fine.

Reply:
What does your implementation look like? The following passes mypy just fine for me:

```python

def available_versions(pretty: bool = True) -> list[str] | None:
    """
    Report the available versions of the `intake.cat.access_nri` catalog.

    Parameters
    ---------
    pretty : bool, optional
        Defines whether to return a pretty print-out of the available versions
        (True, default), or to provide a list of version numbers only (False).
    """
    # Work out where the catalogs are stored
    base_path = _get_catalog_rp()

    # Grab the extant catalog and work out its min and max versions
    try:
        with open(get_catalog_fp()) as cat:
            cat_yaml = yaml.safe_load(cat)
            vers_min = cat_yaml["sources"]["access_nri"]["parameters"]["version"]["min"]
            vers_max = cat_yaml["sources"]["access_nri"]["parameters"]["version"]["max"]
            vers_def = cat_yaml["sources"]["access_nri"]["parameters"]["version"][
                "default"
            ]
    except FileNotFoundError:
        raise FileNotFoundError(f"Unable to find catalog at {get_catalog_fp()}")
    except KeyError:
        raise RuntimeError(f"Catalog at {get_catalog_fp()} not correctly formatted")

    # Grab all the catalog names
    cats = [
        d.name
        for d in base_path.iterdir()
        if re.search(CATALOG_NAME_FORMAT, d.name)
        and d.is_dir()
        and ((d.name >= vers_min and d.name <= vers_max) or d.name == vers_def)
    ]
    cats.sort(reverse=True)

    # Find all the symlinked versions
    symlinks = [s for s in cats if (Path(base_path) / s).is_symlink()]

    symlink_targets = {s: (base_path / s).readlink().name for s in symlinks}

    if pretty:
        for c in cats:
            if c in symlink_targets.keys():
                c += f"(-->{symlink_targets[c]})"
            if c == vers_def:
                c += "*"
            print(c)
        return None

    return cats
```

Reply:
I think going `c` --> `cat` broke/confused things... how odd.